File: java-advent-of-code/src/test/java/advent/twenty_sixteen/Day20FirewallRulesTest.java
package advent.twenty_sixteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day20FirewallRulesTest {
    @Test
    public void checkReal() {
        DailyProblem<Long, Long> dp = new Day20FirewallRules("/twenty_sixteen/Day20-FirewallRules-input.txt", 0, 4294967295l);

        assertEquals(19449262, dp.getPart1Answer().longValue());
        assertEquals(119, dp.getPart2Answer().longValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Long, Long> dp = new Day20FirewallRules("/twenty_sixteen/Day20-FirewallRules-example#1.txt", 0, 9);

        assertEquals(3, dp.getPart1Answer().longValue());
        assertEquals(2, dp.getPart2Answer().longValue());
    }

    @Test
    public void checkExample2() {
        DailyProblem<Long, Long> dp = new Day20FirewallRules("/twenty_sixteen/Day20-FirewallRules-example#2.txt", 0, 19);

        assertEquals(-1, dp.getPart1Answer().longValue());
        assertEquals(0, dp.getPart2Answer().longValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_sixteen/Day20FirewallRulesTest.scala
package advent.twenty_sixteen

import org.scalatest.{FlatSpec, Matchers}

class Day20FirewallRulesTest  extends FlatSpec with Matchers {
  "2016 Day 20 - Input" should "calculate answers" in {
    val dp = new Day20FirewallRules("/twenty_sixteen/Day20-FirewallRules-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day20FirewallRules("/twenty_sixteen/Day20-FirewallRules-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
